src/microsim/schema/simulation.py

```diff
@@ -53,7 +53,7 @@ class Simulation(SimBaseModel):
 
     truth_space: Space
     output_space: Space | None = None
-    sample: Sample
+    samples: Sample | list[Sample]
```

Comment:
```suggestion
    samples: list[Sample]
```

it's best in models to avoid unions (except perhaps in the case of `Optional`).  If one type can serve both purposes, just use that one.  Here we can avoid having to check `if isinstance(samples, list)` everywhere else in the code.  And your field_validator is casting a single object anyway